be/src/exprs/vectorized/binary_function.h

```diff
@@ -222,7 +222,11 @@ class ProduceNullableColumnBinaryFunction {
                     PRODUCE_NULL_FN::template evaluate<LType, RType, TYPE_NULL>(v1, v2));
 
             ColumnPtr data_result = FN::template evaluate<LType, RType, ResultType>(v1, v2);
-            return NullableColumn::create(std::move(data_result), std::move(null_result));
+            if constexpr (pt_is_decimal<ResultType>) {
```

Comment:
If for decimal type, non-nullable column op non-nullable column could produce null result, I think we could use another function  template ?

Reply:
ok

Reply:
commented